src/shirt/string_render.clj

```diff
@@ -52,8 +52,10 @@
 
 (def num-candidates 2500)
 (defn best-partitions [s ^Graphics g width y height]
-  (let [candidates (for [i (range num-candidates)] (candidate-partition s y height width i g))]
-    (first (sort-by (partial candidate-badness g width height) candidates))))
```

Comment:
I was assuming it got GCed when the memoized version of the function goes
out of scope, but if it doesn't, then yeah, shit.

On Thu, Sep 8, 2016, 08:22 Gregor Stocks <notifications@github.com> wrote:

> In src/shirt/string_render.clj
> <https://github.com/GregorStocks/shirt/pull/8#discussion_r78028578>:
>
> > @@ -52,8 +52,10 @@
> >
> >  (def num-candidates 2500)
> >  (defn best-partitions [s ^Graphics g width y height]
> > -  (let [candidates (for [i (range num-candidates)] (candidate-partition s y height width i g))]
> > -    (first (sort-by (partial candidate-badness g width height) candidates))))
>
> we probably don't want memoize as candidate-badness gets called a lot,
> with a lot of different parameters, and I don't know how Graphics work but
> they might be large and if we memoize we can't GC them. Probably
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GregorStocks/shirt/pull/8/files/7e264b8f1904470ba672ca68e3c6daa56a306647#r78028578>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADgeZpaW6WPJ_vIr0RS_oTn8IIj0-sNDks5qoCgggaJpZM4J3fbg>
> .
>